semanticdb/metacp/src/main/scala/scala/meta/internal/javacp/Javacp.scala

```diff
@@ -194,18 +195,29 @@ object Javacp {
             method.signature.params
           }
 
-        val parameterSymbols = params.zipWithIndex.map {
+        val parameterSymbols: List[String] = params.zipWithIndex.map {
           case (param: JavaTypeSignature, i) =>
             val paramName = {
               if (method.node.parameters == null) "param" + i
               else method.node.parameters.get(i).name
             }
             val paramSymbol = Symbols.Global(methodSymbol, d.Parameter(paramName))
+            val isRepeatedType = method.node.access.hasFlag(o.ACC_VARARGS) && i == params.length - 1
+            val tpe =
+              if (isRepeatedType)
+                s.Type(
+                  tag = t.REPEATED_TYPE,
+                  repeatedType = Some(s.RepeatedType(
+                    param.toType(methodScope).typeRef.flatMap(_.typeArguments.headOption)
```

Comment:
this is probably not the most elegant way to achieve this.
I'm essentially converting the type to an `Array[T]`, and then accessing `T`.

Can this be improved?

Reply:
s.Type in general is quite ugly at this point. We have https://github.com/scalameta/scalameta/issues/1262 to track that, but I don't feel we have enough experience with s.Type to design a reasonable API for it. Hence, we're just biting the bullet and writing verbose code for the time being.

Reply:
👍 I'll leave it as it is then